.github/workflows/testing.yml

```diff
@@ -0,0 +1,49 @@
+name: Testing
+# in this file we describe the action to execute for the CI in the dev pipeline
+
+# based on default laravel workflow
+# https://github.com/actions/starter-workflows/blob/master/ci/laravel.yml
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - master
+      - develop
+
+jobs:
+  laravel-tests:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Checkout to current branch
+      run: git fetch && git checkout ${{ github.head_ref }}
+    - name: Copy .env
+      run: php -r "file_exists('.env') || copy('.env.example', '.env');"
+      working-directory: ./src
+    - name: Install Dependencies
+      run: composer install -q --no-ansi --no-interaction --no-scripts --no-suggest --no-progress --prefer-dist
+      working-directory: ./src
+    - name: Generate key
+      run: php artisan key:generate
+      working-directory: ./src
+    - name: Directory Permissions
+      run: chmod -R 777 storage bootstrap/cache
```

Comment:
why?

Reply:
.env has no key by default on this ephemeral machine

Reply:
uh? whats the relation between what you are doing in line 34 (i.e. `chmod -R 777`) and the .env file?

Reply:
uh? whats the relation between what you are doing in line 34 (i.e. chmod -R 777) and the .env file?

Reply:
the key in the .env is generated by doing a php artisan key:generate, why a chmod 777?